crates/uv-auth/src/credentials.rs

```diff
@@ -345,6 +348,126 @@ impl Credentials {
     }
 }
 
+#[derive(Clone, Debug)]
+pub(crate) enum Authentication {
+    /// HTTP Basic or Bearer Authentication credentials.
+    Credentials(Credentials),
+
+    /// AWS Signature Version 4 signing.
+    Signer(DefaultSigner),
+}
```

Comment:
A lot of churn in this PR comes from this piece. We need a new case, which is: a signer that can sign a request (as opposed to `Credentials`, which can be stored in a file, converted into a header, etc.). I initially implemented this as a new variant on the `Credentials` enum, but it made too many invalid states possible (e.g., `to_header_value` had to return an `Option`; the TOML deserialization had to fail). Instead, I added a new type one level higher on the hierarchy which has an `authenticate` method. I think this is much cleaner but does lead to more churn.